From 7d34f85d9845cd62043a205476872d47b839dab0 Mon Sep 17 00:00:00 2001
From: Frajo Haider <f_haider@gmx.at>
Date: Thu, 23 Jan 2020 14:25:53 +0000
Subject: [PATCH 09/12] Revert "linux-user: Call qcrypto_init if not using
 -seed"

This reverts commit a573e9bac6f28cfaa93629b0ebaaa20594365b1d.

Conflicts:
	linux-user/main.c
---
 linux-user/main.c | 21 ++++++++++-----------
 1 file changed, 10 insertions(+), 11 deletions(-)

diff --git a/linux-user/main.c b/linux-user/main.c
index 8ffc525..c60bf17 100644
--- a/linux-user/main.c
+++ b/linux-user/main.c
@@ -43,7 +43,6 @@
 #include "trace/control.h"
 #include "target_elf.h"
 #include "cpu_loop-common.h"
-#include "crypto/init.h"
 
 char *exec_path;
 
@@ -696,17 +695,17 @@ int main(int argc, char **argv, char **envp)
     if (seed_optarg == NULL) {
         seed_optarg = getenv("QEMU_RAND_SEED");
     }
-    {
-        Error *err = NULL;
-        if (seed_optarg != NULL) {
-            qemu_guest_random_seed_main(seed_optarg, &err);
-        } else {
-            qcrypto_init(&err);
-        }
-        if (err) {
-            error_reportf_err(err, "cannot initialize crypto: ");
-            exit(1);
+    if (seed_optarg != NULL) {
+        unsigned long long seed;
+
+        /* This will go away with the last user of rand(). */
+        if (parse_uint_full(seed_optarg, &seed, 0) != 0) {
+            fprintf(stderr, "Invalid seed number: %s\n", seed_optarg);
+            exit(EXIT_FAILURE);
         }
+        srand(seed);
+
+        qemu_guest_random_seed_main(seed_optarg, &error_fatal);
     }
 
     target_environ = envlist_to_environ(envlist, NULL);
-- 
1.8.3-rc3

